src/components/form/Form.js

```diff
@@ -473,11 +473,11 @@ export default class FormComponent extends Component {
   }
 
   hideSubmitButton(component) {
-    const isSubmitButton = (component.type === 'button') &&
-      ((component.action === 'submit') || !component.action);
+    const isSubmitButton = component.type === 'button' && (component.action === 'submit' || !component.action);
 
     if (isSubmitButton) {
       component.hidden = true;
+      component.customConditional = 'show = false';
```

Comment:
What's up with this?

Reply:
In nested forms we hide the submit button (to avoid two submit buttons). Previously, it was enough to attach `component.hidden` to the JSON to both (a) hide the button visually and (b) clear its data from the nested form submission (i.e. `{ data: { submit: true } }`). Now that we've changed clear on hide behavior, we need to preserve backwards-compatible button behavior by hiding it visually _and_ clearing its data, which in this case now means it needs to be conditionally hidden.